src/timepicker/timepicker.js

```diff
@@ -132,21 +66,28 @@ class TimePicker extends React.Component<TimePickerPropsT, TimePickerStateT> {
         steps: steps,
         value: this.props.nullable
           ? undefined
-          : {id: closestStep, label: secondsToLabel(closestStep)},
+          : {
+              id: closestStep,
+              label: this.secondsToLabel(closestStep, undefined),
+            },
       });
       if (this.props.value || (!this.props.nullable && !this.props.value)) {
         this.handleChange(closestStep);
       }
     }
   }
 
-  componentDidUpdate(prevProps: TimePickerPropsT) {
+  componentDidUpdate(prevProps: TimePickerPropsT<T>) {
     const formatChanged = prevProps.format !== this.props.format;
     const stepChanged = prevProps.step !== this.props.step;
+    const adapterChanged = prevProps.adapter !== this.props.adapter;
     if (formatChanged || stepChanged) {
       const steps = this.buildSteps();
       this.setState({steps});
     }
+    if (adapterChanged) {
+      this.dateHelpers = new DateHelpers(this.props.adapter);
+    }
```

Comment:
Maybe this should go before the potential `buildSteps` and `setState` calls? My concern is that the new list would use the old adapter.

Reply:
Yep, that would 100% lead to problems, good catch.